src/hdf5_interface.F90

```diff
@@ -787,6 +791,76 @@ subroutine hdf5_read_string(group, name, buffer, length)
   end subroutine hdf5_read_string
 
 !===============================================================================
+! HDF5_WRITE_STRING_1DARRAY writes string data
+!===============================================================================
+
+  subroutine hdf5_write_string_1Darray(group, name, buffer, length, rank)
```

Comment:
Oh, these string handling routines were intended to write out the "label" on given to tallies in the talies.xml file. In addition, I have the "python-api-labels" which includes a new optional "label" XML tag for materials, surfaces, cells, and lattices. The "label" is read in from XML, assigned to the appropriate data structure, and written out to the ``summary.h5`` file (if the user requested it). The Python API on that branch then links the labels back to the appropriate Python objects when a ``openmc.Summary`` object is instantiated with ``summary.h5``. Likewise, all of the labels are connected to the appropriate counterparts if a user calls the ``Statepoint.link_with_summary(...)`` routine. 

In any case, these routines are useful for the labeling functionality. If you think that allowing users to label their objects is an option you'd like to incorporate, I can merge that branch into this one for this PR.